apps/next/app/api/deploy/[tag]/status/route.ts

```diff
@@ -46,12 +55,33 @@ const mockGetDeploymentStatus = async (deploymentTag: string): Promise<Deploymen
   });
 };
 
-async function* makeIterator() {
-  yield encoder.encode('Deployment in progress...')
-  await sleep(5000)
-  yield encoder.encode('Cleaning deployment files...')
-  await sleep(5000)
-  yield encoder.encode('Deployed...')
+async function* makeIterator(buildTag: string) {
+  yield encoder.encode('Planning build...');
+  const { stdout: stdoutJson } = await exec('nixpacks plan ./temp');
+  const plan = JSON.parse(stdoutJson) as NixpackPlan;
+  // Filtering out 'npm-9_x' because for some reason it's not listed on:
+  // https://search.nixos.org/packages
+  plan.phases.setup.nixPkgs =
+    plan.phases.setup.nixPkgs.filter(nixPackage => nixPackage !== 'npm-9_x');
+  writeFileSync('./temp/build-plan.json', JSON.stringify(plan, null, 2), "utf8");
```

Comment:
instead of using ./temp can you instead use https://www.npmjs.com/package/tmp and `tmp.dirSync()`?

this way concurrent calls to this function (e.g. two requests in flight at the same time) don't interfere with each other

Reply:
I've done this, but I had to modify it a bit. Since the "up" command actually uses two rest endpoints (upload and status), using a random dir name will require more work as both endpoints need to know its value. So, I modified your comment to be that each upload is extracted into a different dir with names like `up_<tag>`, and that dir is placed in a common `ups` directory that will contain all deployments.

LMK what you think.